src/android/Device.java

```diff
@@ -59,31 +69,118 @@ public void initialize(CordovaInterface cordova, CordovaWebView webView) {
+            addToReturnValue("version", this.getOSVersion());
+            addToReturnValue("platform", this.getPlatform());
+            addToReturnValue("model", this.getModel());
+            addToReturnValue("manufacturer", this.getManufacturer());
+            addToReturnValue("isVirtual", this.isVirtual());
+
+            // Due to privacy restrictions ALL devices running on Android 10 or higher
+            // "getSerialNumber()" will just return the string "unknown" instead of the real device serial number ...
+            addToReturnValue("serial", this.getSerialNumber());
+
+            // ... luckily, at least zebra devices offer a way to deliver their serial number on Android 10 or higher.
+            // So in case the device is a zebra device running on Android >= 10
+            // we are trying to (asynchronously) access the serial number via a DeviceIdentifiersWrapper
+            // and replace the "unknown" serial number in the PluginResult with the real serial number.
+            if (this.isZebraDevice() && android.os.Build.VERSION.SDK_INT >= 29) {
+                this.triggerZebraSNRetrieval(this.cordova.getContext());
+            } else {
+                // in any other case, we instantly return the collected device values back to the caller
+                // (serial number will be "unknown")
+                finishPluginCallback();
+            }
+
+            return true;
         }
         else {
             return false;
         }
-        return true;
+
     }
 
+    /**
+     * Helper function to store values in PluginResult (= JSONObject)
+     *
+     * @param key            The key (e.g. "uuid")
+     * @param value          The value (e.g. "61363bf5509a717f")
+     * @return
+     */
+    private void addToReturnValue(String key, Object value)  {
+        try {
+            this._returnValue.put(key, value);
+        } catch(Exception e) {
+            Log.e("addToReturnValue", e.getMessage());
+        }
+    }
+
+    /**
+     * Helper function to return PluginResult to the caller
+     *
+     * @return
+     */
+    private void finishPluginCallback() {
+        this._callbackContext.success(this._returnValue);
+    }
+
+    /**
+     * Retrieves the serial number of a zebra device using a DeviceIdentifiersWrapper
+     *
+     * For more information see:
+     * https://github.com/ZebraDevs/DeviceIdentifiersWrapper
+     *
+     * @return
+     */
+    private void triggerZebraSNRetrieval(Context context)
+    {
+        DIHelper.getSerialNumber(context, new IDIResultCallbacks() {
+            @Override
+            public void onSuccess(String serialNumber) {
+                // replace "unknown" with the real serial number in the PluginResult
+                addToReturnValue("serial", serialNumber);
+                finishPluginCallback();
+            }
+
+            @Override
+            public void onError(String message) {
+                Log.e("triggerZebraSNRetrieval", message);
```

Comment:
The first parameter should be a tag unique to this module. I'd recommend adding a member variable for this purpose:

```java
private static final String TAG = Device.class.getCanonicalName();
``` 

Also, you should add more context to the message:

```suggestion
                Log.e(TAG, "error while retrieving Zebra serial number: " + message);
```